services/brig/src/Brig/Provider/API.hs

```diff
@@ -495,20 +517,28 @@ addServiceH (pid ::: req) = do
     sid      <- DB.insertService pid name summary descr baseUrl token pk fp assets tags
 
     let rstoken = maybe (Just token) (const Nothing) (newServiceToken new)
-    return $ setStatus status201
-           $ json (NewServiceResponse sid rstoken)
+    return $ NewServiceResponse sid rstoken
 
 listServicesH :: ProviderId -> Handler Response
-listServicesH pid = json <$> DB.listServices pid
+listServicesH pid = json <$> listServices pid
+
+listServices :: ProviderId -> Handler [Service]
+listServices = DB.listServices
 
 getServiceH :: ProviderId ::: ServiceId -> Handler Response
 getServiceH (pid ::: sid) = do
-    s <- DB.lookupService pid sid >>= maybeServiceNotFound
-    return (json s)
+    json <$> getService pid sid
+
+getService :: ProviderId -> ServiceId -> Handler Service
+getService pid sid = do
+    DB.lookupService pid sid >>= maybeServiceNotFound
```

Comment:
> if you leave it for now, we can do it later.

agreed.  i think this would have kept me from finishing even brig, and i still need to get the others done.  effects are a much bigger topic.